src/components/ProjectSearch.js

```diff
@@ -123,6 +123,13 @@ export class ProjectSearch extends Component<Props, State> {
     shortcuts.off("Enter", this.onEnterPress);
   }
 
+  componentDidUpdate(prevProps: Props) {
+    // If the query changes in redux, also change it in the UI
+    if (prevProps.query !== this.props.query) {
+      this.setState({ inputValue: this.props.query });
```

Comment:
I know generally deriving state from props is considered bad practice but this was the simplest way I came up with to keep the query string from persisting in the UI between navigations.  Let me know if you have other ideas of how to do this.  Or a case could me made to keep the query there if it is likely someone would want to search for the same term on a different page.

Reply:
@jasonLaster Do you have a suggestion here?

Reply:
I think that `props.query` is the query used for the current search results and `state.inputValue` is the current value in the UI input box so the state is not quite redundant.  When the user starts typing in the input box, `inputValue` changes but the `query` doesn't change until the user executes the search.

Reply:
ahh thanks